code/modules/hazmat/crystal.dm

```diff
@@ -0,0 +1,42 @@
+#define CRYSTAL_SHAPE_SQUARE "square"
+#define CRYSTAL_SHAPE_CIRCLE "circle"
+#define CRYSTAL_SHAPE_TRIANGLE "triangle"
+#define CRYSTAL_SHAPE_CYLINDER "cylinder"
+
+#define CRYSTAL_SIZE_SMALL "small"
+#define CRYSTAL_SIZE_MEDIUM "medium"
+#define CRYSTAL_SIZE_LARGE "large"
+
+#define CRYSTAL_COLOR_RED "#FF0000"
+#define CRYSTAL_COLOR_BLUE "#00FF00"
+#define CRYSTAL_COLOR_GREEN "#0000FF
+
+/obj/item/crystal
+	name = "anomalous crystal"
+	desc = "Do you know who ate all the donuts?"
+	icon = 'icons/obj/hazmat/crystals.dmi'
+	icon_state = "crystal_square"
+	var/crystal_shape = CRYSTAL_SHAPE_SQUARE
+	var/crystal_size = CRYSTAL_SIZE_SMALL
```

Comment:
should probably default to MEDIUM, just to keep weird stuff from happening if generate_crystal_visuals() gets called twice on the same crystal somehow

Reply:
For example, crystals generated when the crystal maker is attacked by a crystal (and replaces that crystal with a new one) are half the size that they "should" be, because Initialize() calls generate_crystal_visuals() before the new crystal receives its updated crystal_size value (which means that the size that it's called with is CRYSTAL_SIZE_SMALL, which shrinks the size of the crystal's sprite), then generate_crystal_visuals() gets called a second time with the "real" crystal_size value later.

Am I making sense here, or do I need to explain what's causing that bug in a different way?